driver-async/src/main/com/mongodb/async/client/ClientSession.java

```diff
@@ -33,6 +33,16 @@
      */
     boolean hasActiveTransaction();
 
+    /**
+     *  Notify the client session that a message has been sent.
+     *  <p>
+     *      For internal use only
+     *  </p>
+     *
+     * @return true if this is the first message sent, false otherwise
+     */
+    boolean notifyMessageSent();
```

Comment:
I don't see a way around adding a new public method to ClientSession

Reply:
Other options I considered for this method:

1. Return false if this is the first message sent.  
2. Returning void and adding another method like `isMessageSent`

It's for internal use only (if the application calls it before a message has been sent, all hell will break loose as the first command sent to the server will be incorrect, but that was true already with ServerSession.advanceStatementId), so I didn't stress too much about it in the end.

Reply:
Ack.